pkgs/games/openra/mod.nix

```diff
@@ -0,0 +1,86 @@
+# Based on: https://build.opensuse.org/package/show/home:fusion809/openra-ura
+{ stdenv
+, packageAttrs
+, patchEngine
+, mkdirp
+, wrapLaunchGame
+, mod
+, engine
+}:
+
+with stdenv.lib;
+
+stdenv.mkDerivation (recursiveUpdate packageAttrs {
+  name = "openra-${mod.name}-${mod.version}";
+
+  srcs = [
+    mod.src
+    engine.src
+  ];
+
+  sourceRoot = ".";
+
+  postUnpack = ''
+    mv ${engine.src.name} ${mod.src.name}
```

Comment:
So you can't build engine separately?

Reply:
I asked this too, but my own attempts failed. From what I understand the mods need to have the source of the engine available, and since every mod uses a specific engine so far, there is also not much gained with separately building the engine anyway.